src/cdomains/intDomain.ml

```diff
@@ -1826,9 +1876,470 @@ module Enums : S with type int_t = BigInt.t = struct
+    | _, x when leq zero x -> top () (*TODO warn about undefined behaviour when x equals zero*)
+    | Some(c1, m1), Some(c2, m2) when not no_ov && m2 = Ints_t.zero && c2 = Ints_t.neg Ints_t.one -> top ()
+    | Some(c1, m1), Some(c2, m2) when m1 =: Ints_t.zero && m2 = Ints_t.zero -> Some(c1 /: c2, Ints_t.zero)
+    | Some(c1, m1), Some(c2, m2) when m2 =: Ints_t.zero ->  if (c2 |: m1) && (c2 |: c1) then Some(c1 /: c2, m1 /: c2) else top ()
+    | _, _ -> top ()
+
+
+  let div ?no_ov ik x y =
+    let res = div ?no_ov ik x y in
+    if M.tracing then
+      M.trace "mything" "Cong. div : %a %a -> %a \n" pretty x pretty y pretty
+        res ;
+    res
+
+  let ne ik i1 i2 = if meet ik i1 i2 = None then of_bool ik true else top_bool
+
+  let eq ik (i1: t) (i2: t) = if meet ik i1 i2 <> None then top_bool else of_bool ik false
+
+  let ge ik x y = match x, y with
+    | None, None -> bot_of ik
+    | None, _ | _, None -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | Some (c1,m1), Some (c2,m2) -> if (m1 =: Ints_t.zero) && (m2 =: Ints_t.zero) then
+        if c1 >=: c2 then of_bool ik true else of_bool ik false
+      else top_bool
+
+  let ge ik x y =
+    let res = ge ik x y in
+    if M.tracing then  M.trace "mything" "Cong. greater or equal : %a %a becomes %a \n" pretty x pretty y pretty res;
+    res
+
+  let le ik x y = match x, y with
+    | None, None -> bot_of ik
+    | None, _ | _, None -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | Some (c1,m1), Some (c2,m2) -> if (m1 =: Ints_t.zero) && (m2 =: Ints_t.zero) then
+        if c1 <=: c2 then of_bool ik true else of_bool ik false
+      else top_bool
+
+  let le ik x y =
+    let res = le ik x y in
+    if M.tracing then  M.trace "mything" "Cong. less or equal : %a %a becomes %a \n" pretty x pretty y pretty res;
+    res
+
+  let gt ik x y = match x, y with
+    | None, None -> bot_of ik
+    | None, _ | _, None -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | Some (c1,m1), Some (c2,m2) -> if (m1 =: Ints_t.zero) && (m2 =: Ints_t.zero) then
+        if c1 >: c2 then of_bool ik true else of_bool ik false
+      else top_bool
+
+  let gt ik x y =
+    let res = gt ik x y in
+    if M.tracing then  M.trace "mything" "Cong. greater than : %a %a becomes %a \n" pretty x pretty y pretty res;
+    res
+
+  let lt ik x y = match x, y with
+    | None, None -> bot_of ik
+    | None, _ | _, None -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | Some (c1,m1), Some (c2,m2) -> if (m1 =: Ints_t.zero) && (m2 =: Ints_t.zero) then
+        if c1 <: c2 then of_bool ik true else of_bool ik false
+      else top_bool
+
+  let lt ik x y =
+    let res = lt ik x y in
+    if M.tracing then  M.trace "mything" "Cong. less than : %a %a becomes %a \n" pretty x pretty y pretty res;
+    res
+
+  let invariant c x = failwith "unimplemented"
```

Comment:
Please patch these invaraints to return `None` (for now, actually it would be good to implement them). 
Otherwise, I can't run your tests with SV-COMP